### DISCUSSION TITLE
Potencial to receive data from other TX

### DISCUSSION THREAD

**Discussion by TonyInvent** [ExpressLRS/ExpressLRS#856] [Q&A]
Buffer[0] = RC_DATA_PACKET & 0b11;
  Buffer[1] = ((crsf->ChannelDataIn[0]) >> 3);
  Buffer[2] = ((crsf->ChannelDataIn[1]) >> 3);
  Buffer[3] = ((crsf->ChannelDataIn[2]) >> 3);
  Buffer[4] = ((crsf->ChannelDataIn[3]) >> 3);
  Buffer[5] = ((crsf->ChannelDataIn[0] & 0b110) << 5) |
                           ((crsf->ChannelDataIn[1] & 0b110) << 3) |
                           ((crsf->ChannelDataIn[2] & 0b110) << 1) |
                           ((crsf->ChannelDataIn[3] & 0b110) >> 1);

  uint16_t crc = ota_crc.calc(Radio.TXdataBuffer, 7, CRCInitializer);
  Radio.TXdataBuffer[0] |= (crc >> 6) & 0b11111100;
  Radio.TXdataBuffer[7] = crc & 0xFF;

since data in TX has no TX ID info. 
For the RX side, how to avoid data from other TXs being received?

**Comment by pkendall64** [comment]
The CRC initialiser has a part of the UID (generated from the passphrase) so that means you might have a collision 1 in 65536 times as is uses the final 2 bytes of the UID.

**Reply by TonyInvent** [reply]
@pkendall64 got you. 
uint16_t CRCInitializer = (UID[4] << 8) | UID[5];
btw, thank you very much. You answer give me much more confidence on ExpressLRS.
